### PR TITLE
Fixed hostname update bug

### PR DESCRIPTION
**Pull request by ashok133**
Once WI is applied, the updated KFP dashboard is fetched and pipeline is deployed on the new nodes. 

Caveat: The user still needs to update the hostname in `config.yaml` for subsequent runs.

### CODE REVIEW

File: ml_pipeline_gen/pipelines.py [hunk 1]
```diff
@@ -151,21 +153,40 @@ class KfpPipeline(BasePipeline):
...
+            "bin/wi_setup.sh",
+            model.project_id,
+            model.cluster_name,
+            model.cluster_zone,
+            # TODO(ashokpatelapk): Check if namespace can be a config var.
+            "default"
+        ])
+
+    def update_hostname(self):
+        """Returns Hostname (URL) of KFP cluster in current kube context."""
```
Comment (TheMichaelHu) [inline ml_pipeline_gen/pipelines.py]: I don't see any return statement.

File: ml_pipeline_gen/pipelines.py [hunk 2]
```diff
@@ -151,21 +153,40 @@ class KfpPipeline(BasePipeline):
 
     def __init__(self, model=None, config=None):
         super().__init__(model, config)
-        self.setup_auth()
+        if not self.check_cluster_label("mlpg_wi_auth"):
+            self.setup_auth()
+            self.update_hostname()
 
     def setup_auth(self):
         """Calls shell script to verify required auth for KFP cluster."""
```
Comment (TheMichaelHu) [inline ml_pipeline_gen/pipelines.py]: nit: I think we need documentation on what the script is doing either in this docstring or at the top of the shell script. Preferably both.

**Review by TheMichaelHu — APPROVED** [review]
Some documentation nits. LGTM otherwise

File: ml_pipeline_gen/pipelines.py (hunk 2 above)
Comment (ashok133) [reply ml_pipeline_gen/pipelines.py]: The `wi_auth.sh` script already has documentation, will flesh out the docstring to be more descriptive.

Comment (TheMichaelHu) [reply ml_pipeline_gen/pipelines.py]: oh yup, you're right.

**Pull request merged**